src/main/java/com/booksaw/betterTeams/team/storage/convert/YamlToSql.java

```diff
@@ -153,25 +154,28 @@ protected void convert() {
 
 	public static void pack(String sourceDirPath, String zipFilePath) throws IOException {
 		File f = new File(zipFilePath);
-		Path p;
-		if (f.exists()) {
-			p = Paths.get(zipFilePath);
-		} else {
-			p = Files.createFile(Paths.get(zipFilePath));
-		}
+
+		Path p = Paths.get(zipFilePath);
+
+		if (!f.exists()) {
+            Files.createFile(p);
+        }
 
 		try (ZipOutputStream zs = new ZipOutputStream(Files.newOutputStream(p))) {
 			Path pp = Paths.get(sourceDirPath);
-			Files.walk(pp).filter(path -> !Files.isDirectory(path)).forEach(path -> {
-				ZipEntry zipEntry = new ZipEntry(pp.relativize(path).toString());
-				try {
-					zs.putNextEntry(zipEntry);
-					Files.copy(path, zs);
-					zs.closeEntry();
-				} catch (IOException e) {
-					System.err.println(e);
-				}
-			});
+
+			try (Stream<Path> paths = Files.walk(pp)) {
+				paths.filter(path -> !Files.isDirectory(path)).forEach(path -> {
+					ZipEntry zipEntry = new ZipEntry(pp.relativize(path).toString());
+					try {
+						zs.putNextEntry(zipEntry);
+						Files.copy(path, zs);
+						zs.closeEntry();
+					} catch (IOException e) {
+						System.err.println(e);
+					}
+				});
+			}
```

Comment:
should we not have some error handing for the additional try here?

Reply:
I honestly let this bubble up...
It'll be handled in YamlToSql::convert. But the try-handler is actually not a try-except, but a handler to close resources here.
try-with-resources: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
vs
try-block: https://docs.oracle.com/javase/tutorial/essential/exceptions/try.html